ui/raidboss/data/05-shb/raid/e8s.js

```diff
@@ -384,6 +384,95 @@
         }
       },
     },
+    {
+      id: 'E8S Mirror, Mirror',
+      regex: Regexes.startsUsing({ source: 'Shiva', id: '4D5A', capture: false }),
+      regexDe: Regexes.startsUsing({ source: 'Shiva', id: '4D5A', capture: false }),
+      regexFr: Regexes.startsUsing({ source: 'Shiva', id: '4D5A', capture: false }),
+      regexJa: Regexes.startsUsing({ source: 'シヴァ', id: '4D5A', capture: false }),
+      regexCn: Regexes.startsUsing({ source: '希瓦', id: '4D5A', capture: false }),
+      regexKo: Regexes.startsUsing({ source: '시바', id: '4D5A', capture: false }),
+      condition: function(data) {
+        data.mirrorCast = data.mirrorCast || 0;
+        data.mirrorCast++;
+        data.isHallowedWingsMirrorMirror = data.mirrorCast === 3;
+
+        return data.isHallowedWingsMirrorMirror;
+      },
+      delaySeconds: 12,
+      promise: function(data) {
+        let p = new Promise(async (res) => {
+          const shivaLocaleNames = {
+            en: 'Shiva',
+            de: 'Shiva',
+            fr: 'Shiva',
+            ja: 'シヴァ',
+            cn: '希瓦',
+            ko: '시바',
+          };
+
+          let combatantData = await window.callOverlayHandler({
+            call: 'getCombatants',
+            names: [shivaLocaleNames[data.lang]],
+          });
+
+          // if we could not retrieve combatant data, the
+          // trigger will not work, so just resume promise here.
+          if (!(combatantData !== null &&
+            combatantData.combatants &&
+            combatantData.combatants.length)) {
+            data.safeZone = null;
+            res();
+            return;
+          }
+
+          // if we dont have information about which wing was
+          // casted, then the trigger will not work (safety measure)
+          if (!(data.leftWing || data.rightWing)) {
+            data.safeZone = null;
+            res();
+            return;
+          }
+
+          // we want to inspect the active shiva (it's the one with the lowest id)
+          let shivaCombatant =
+            combatantData.combatants.sort((a, b) => b.ID - a.ID).pop();
+
+          // we have 2 inputs: shiva faces north/south
+          //                   left/right wing is glowing
+          let ttsNumbers = '';
+          let sideResponse = {};
+
+          if (data.leftWing) {
+            sideResponse = Responses.goRight().alertText;
+            if (shivaCombatant.Heading > 3)
+              ttsNumbers = '1 4 ';
+          }
+
+          if (data.rightWing) {
+            sideResponse = Responses.goLeft().alertText;
+            if (shivaCombatant.Heading > 3)
+              ttsNumbers = '2 3 ';
+          }
```

Comment:
Might be a good idea to `Math.abs` the Heading value before comparison, e.g. `if (Math.abs(shivaCombatant.Heading) > 3)`. Might not be 100% necessary but heading ranges from -π to π (-3.14... to 3.14...) and the direct north/south cardinals can vary depending on rounding errors.

Reply:
Nice catch, got that in the triggernometry trigger but forgot that to implement here aswell.